syntax = "proto3";

package distributed_banking;

message BankingOperationRequest {
    int32 id = 1;
    string type = 2;
    repeated CustomerRequest customer_requests = 3;
}

message BankingOperationResponse {
    repeated EventResult event_result = 1;
}

message CustomerRequest {
    int32 customer_request_id = 1;
    string interface = 2;
    int32 logical_clock = 3;
    int32 money = 4;
}

message EventResult {
    int32 id = 1;
    int32 customer_request_id = 2;
    string type = 3;
    int32 logical_clock = 4;
    string interface = 5;
    string comment = 6;
}

service BankingService {
    rpc MsgDelivery (BankingOperationRequest) returns (BankingOperationResponse);
}
